src/Stratis.Bitcoin/Connection/ConnectionManager.cs

```diff
@@ -264,13 +268,69 @@ public void Dispose()
             foreach (NetworkPeerServer server in this.Servers)
                 server.Dispose();
 
+            this.connectedPeersQueue.Dispose();
+
             this.networkPeerDisposer.Dispose();
         }
 
         /// <inheritdoc />
         public void AddConnectedPeer(INetworkPeer peer)
         {
             this.connectedPeers.Add(peer);
+            this.connectedPeersQueue.Enqueue(peer);
```

Comment:
Why asyncqueue this operations should be quite fast right?

Reply:
To avoid race conditions. I don't want to use locks here but also don't want peers to be added to the same collection right after I've checked against it or in the middle of it.

Reply:
ah right yeah and asyncqueue is single threaded...  elegant solution actually.